metadata-service/auth-impl/src/main/java/com/datahub/authentication/authenticator/AuthenticatorChain.java

```diff
@@ -57,12 +59,16 @@ public Authentication authenticate(@Nonnull final AuthenticatorContext context)
         throw e;
       } catch (Exception e) {
         // Log as a normal error otherwise.
-        log.error(String.format(
+        authenticationFailures.add(new Pair<>(authenticator.getClass().getCanonicalName(), e.getMessage()));
+        log.debug(String.format(
             "Caught exception while attempting to authenticate request using Authenticator %s",
             authenticator.getClass().getCanonicalName()), e);
       }
     }
     // No authentication resolved. Return null.
+    if (!authenticationFailures.isEmpty()) {
+      log.error("Authentication chain failed to resolve a valid authentication. Errors: {}", authenticationFailures);
```

Comment:
We can actually make this a WARN -- It's not really an ERROR per say... What do you think? Let me know and I can inline the change

Reply:
Done :)